File: sdk-mobile/ios/AgeKeySwift/Sources/AgeKeySwift/AgeKeyConfig.swift
import Foundation

public enum AgeKeyEnvironment: Sendable, Equatable {
    case production
    case staging
    case custom(baseURL: URL)

    public var baseURL: URL {
        switch self {
        case .production:
            return URL(string: "https://api.agekey.com.br")!
        case .staging:
            return URL(string: "https://staging.agekey.com.br")!
        case .custom(let baseURL):
            return baseURL
        }
    }
}

public struct AgeKeyConfig: Sendable {
    public let apiKey: String
    public let environment: AgeKeyEnvironment
    public let defaultLocale: String

    public init(
        apiKey: String,
        environment: AgeKeyEnvironment = .production,
        defaultLocale: String = "pt-BR"
    ) {
        self.apiKey = apiKey
        self.environment = environment
        self.defaultLocale = defaultLocale
    }
}

File: sdk-mobile/android/agekey-android/agekey/src/main/java/com/ecadigital/agekey/AgeKeyConfig.kt
package com.ecadigital.agekey

data class AgeKeyConfig(
    val apiKey: String,
    val environment: AgeKeyEnvironment = AgeKeyEnvironment.Production,
    val defaultLocale: String = "pt-BR"
)

sealed class AgeKeyEnvironment(val baseUrl: String) {
    data object Production : AgeKeyEnvironment("https://api.agekey.com.br")
    data object Staging : AgeKeyEnvironment("https://staging.agekey.com.br")
    data class Custom(val url: String) : AgeKeyEnvironment(url)
}
